app/services/oidc_jwt_token_validate.rb

```diff
@@ -0,0 +1,57 @@
+class OidcJwtTokenValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def valid_jwt?
+    valid_signature? && type_bearer? && valid_aud?
+  rescue JWT::InvalidAudError
+    logger.error "OidcJwtToken SSO: Invalid token."
+    false
+  rescue JWT::ImmatureSignature
+    logger.error "OidcJwtToken SSO: JWT is for the future."
+    false
+  rescue JWT::ExpiredSignature
+    logger.error "OidcJwtToken SSO: Signature has expired."
+    false
+  rescue JWT::DecodeError
+    logger.error "OidcJwtToken SSO: Failed to decode JWT."
+    false
+  end
+
+  private
+
+  def generate_jwks_uri
+    response = RestClient::Request.execute(
+      :url => Setting['oidc_jwks_url'],
+      :method => :get,
+      :headers => { content_type: 'application/x-www-form-urlencoded' },
```

Comment:
As far as I know this header mainly used  in POST method calls. 
So, just for curiosity I would like to know reason why it is used here in GET request?

Reply:
Good catch, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type.

I'm also wondering if this should have been `'content-type':`.

Reply:
It would be rather better to not have a `content-type` because as i check `RestClient::Request` sets the header to `:content_type=>"application/json"` by default.

Reply:
Yes, remove header value. Request will work as it is. :+1: